db/workflow.go

```diff
@@ -109,9 +109,10 @@ func insertActionList(ctx context.Context, db *sql.DB, yamlData string, id uuid.
 
 		workerID, err := getWorkerID(ctx, db, task.WorkerAddr)
 		if err != nil {
+			if errors.Cause(err) == sql.ErrNoRows {
+				return fmt.Errorf("hardware mentioned with reference %s not found", task.WorkerAddr)
```

Comment:
why not `errors.WithMessage() here instead of dropping the stack frame and all the goodness from errors.Wrap?